File: ABC/351-400/383/b.py
def main():
    h,w,d = map(int,input().split())
    s = [input() for _ in range(h)]
    ans = 0
    for i1 in range(h):
        for j1 in range(w):
            for i2 in range(h):
                for j2 in range(w):
                    if i1 == i2 and j1 == j2:
                        continue
                    
                    if s[i1][j1] == '#' or s[i2][j2] == '#':
                        continue
                    
                    cnt = 0
                    for i in range(h):
                        for j in range(w):
                            if s[i][j] != '.':
                                continue
                            if abs(i-i1) + abs(j-j1) <= d or abs(i-i2) + abs(j-j2)<=d:
                                cnt += 1
                    
                    ans = max(ans,cnt)
    
    print(ans)
    

if __name__ == "__main__":
    main()

File: ABC/351-400/383/b.cpp
#include <bits/stdc++.h>
using namespace std;
#define rep(i,n) for(int i=0;i<n;++i)
const int di[] = {1,1,1,0,0,-1,-1,-1};
const int dj[] = {1,-1,0,1,-1,1,-1,0};
int main(){
    int h,w,d;
    cin >> h >> w >> d;
    vector<string> s(h);
    rep(i,h) cin >> s[i];
    int ans = 0;
    rep(i1,h){
        rep(j1,w){
            rep(i2,h){
                rep(j2,w){
                    if(s[i1][j1] == '#' || s[i2][j2] == '#') continue;
                    if(i1 == i2 && j1 == j2) continue;
                    int now = 0;
                    rep(i,h)rep(j,w){
                        if(s[i][j] == '#') continue;
                        if(abs(i-i1) + abs(j-j1) <= d || abs(i-i2)+abs(j-j2) <= d) now++;
                    }
                    ans = max(ans,now);
                }
            }
        }
    }
    cout << ans << endl;
    return 0;
}